### PR TITLE
HTTP(S) API connection plugin

### PR DESCRIPTION
**Pull request by Qalthos**
##### SUMMARY

implements ansible/proposals#103

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

httpapi

##### ANSIBLE VERSION

```
2.6
```

### CODE REVIEW

File: lib/ansible/plugins/connection/httpapi.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+DOCUMENTATION = """
+    author: Ansible Networking Team
+    connection: httpapi
+    short_description: Use httpapi to run command on Arista EOS devices using httpapi
+    description:
+        - This plugin actually forces use of 'local' execution but using paramiko to establish a remote ssh shell on the appliance.
+        - Also this plugin ignores the become_method but still uses the becoe_user and become_pass to
+          do privilege escalation.
+    version_added: "2.5"
```
Comment (gundalow) [inline lib/ansible/plugins/connection/httpapi.py]: 2.6

File: lib/ansible/plugins/connection/httpapi.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
+# (c) 2018 Red Hat Inc.
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+    author: Ansible Networking Team
+    connection: httpapi
+    short_description: Use httpapi to run command on Arista EOS devices using httpapi
```
Comment (gundalow) [inline lib/ansible/plugins/connection/httpapi.py]: Is this Arista specific?

File: lib/ansible/plugins/connection/httpapi.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    description:
+        - This plugin actually forces use of 'local' execution but using paramiko to establish a remote ssh shell on the appliance.
```
Comment (gundalow) [inline lib/ansible/plugins/connection/httpapi.py]: Do you think talking about `local` may confuse people?
user detail vs implementation detail ?

File: lib/ansible/plugins/connection/httpapi.py [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+        - Also this plugin ignores the become_method but still uses the becoe_user and become_pass to
```
Comment (gundalow) [inline lib/ansible/plugins/connection/httpapi.py]: becoe_user -> become_user

File: lib/ansible/plugins/action/eos.py [hunk 5]
```diff
@@ -43,7 +43,7 @@ def run(self, tmp=None, task_vars=None):
 
         socket_path = None
 
-        if self._play_context.connection == 'network_cli':
+        if self._play_context.connection in ('network_cli', 'httpapi'):
```
Comment (trishnaguha) [inline lib/ansible/plugins/action/eos.py:46]: @Qalthos Do we want to add this check in action/net_base also?

Comment (Qalthos) [reply lib/ansible/plugins/action/eos.py:46]: It isn't necessary, and there isn't really an analogue to this check in net_base now, so I don't think so.

**Comment by Qalthos** [comment]
!needs_rebase

**Pull request merged**